src/Detector/Rendering.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Utopia\Detector\Detector;
+
+use Utopia\Detector\Detection\Rendering as RenderingDetection;
+use Utopia\Detector\Detection\Rendering\SSG;
+use Utopia\Detector\Detection\Rendering\SSR;
+use Utopia\Detector\Detector;
+
+class Rendering extends Detector
+{
+    public function __construct(protected array $inputs, protected string $framework) {}
+
+    public function detect(): RenderingDetection
+    {
+        if (! isset(SSR::FRAMEWORK_FILES[$this->framework])) {
+            throw new \InvalidArgumentException("Unsupported framework: {$this->framework}");
```

Comment:
I dont think it's typical to use PHP custom exception types.
Let's check with Christy if we want  generic exception here, or a custom one defined by library.

Reply:
@christyjacob4 need your suggestion here